### ISSUE TITLE
Broken compatibility with nix-build

### ISSUE THREAD

**Issue by balsoft** [nmattia/niv#154]
With previous (v9) `sources.nix` I was able to just `nix-build nix/sources.nix -A nixpkgs` to get `result` with nixpkgs. If I do that now, by default I don't get a result. That's because now it uses `builtins_fetchTarball` instead of `fetchzip` by default. A solution is to set `builtins: "false"` in `sources.json`. I wonder if it's better to use the non-builtin version by default or somehow make `nix-build` work with a builtin version.

**Comment by michaelpj** [comment]
That seems very weird, given that the builtin tarball fetcher shouldn't have any dependencies. What error do you get exactly?

**Comment by balsoft** [comment]
No error, there's just no `result` symlink (as if the derivation that nix-build tries to build doesn't have any outputs). If I set `-a builtin=false` then it works. Maybe we can set the builtin to false by default to keep the compatibility with people's workflow? This affected at least two people and also my company. I've also spent more time than I should've debugging this :)

**Comment by michaelpj** [comment]
Apparently `builtins.fetchTarball` doesn't evaluate to a derivation, but just gives you a store path. That's... odd. I guess it's because it happens at eval time.

The builtin fetcher is usually better, so this is a bit annoying.

**Comment by balsoft** [comment]
Yes, I came to the same results while debugging, but it's unclear how to fix it. I guess we can create a hack so that it's a derivation instead.

**Comment by nmattia** [comment]
Wow, that's an unexpected use case! Out of curiosity, what do you do with the `result`?

I really can't remember if I meant to use the builtins by default after the 5 -> 6 update. Maybe we should just revert to using `fetchzip`.

**Comment by balsoft** [comment]
> Out of curiosity, what do you do with the result

Mostly just `NIX_PATH=nixpkgs=./result` . I guess I can just add `builtin=false` to all packages, or `ln -s $(nix eval -f ./nix/sources.nix nixpkgs) ./result` or just `NIX_PATH=nixpkgs=$(nix eval -f ./nix/sources.nix nixpkgs)` or something like that.

**Comment by nmattia** [comment]
I think that's the safest thing to do for now. I intend to rework the `sources.nix` to allow passing `nixpkgs`, and also returning `pkgs`, so setting `builtin=false` is the best longterm solution.

I'll keep the use case in mind, it's pretty interesting. So do you use `<nixpkgs>` in nix builds than don't use the `sources.nix`? or in `nix-shell`?
